docs/content/design/proxy.md

```diff
@@ -62,3 +62,25 @@ Private module filters are string globs that tell the Athens proxy what is a pri
 ### Exclude Lists for Public Modules
 
 Exclude lists for public modules are also globs that tell the Athens proxy what modules it should never download from any upstream proxy. For example, the string `github.com/arschles/**` tells the Athens proxy to always return `404 Not Found` to clients.
+
+## Catalog Endpoint
+
+The proxy provides a `/catalog` service endpoint to fetch all the modules and version contained in the local storage.
+The endpoint accepts a continuation token and a page size parameter in order to provide paginated results.
+
+A query is of the form
+
+`https://proxyurl/catalog?token=foo&limit=47`
+
+Where token is an optional continuation token and limit is the desired size of the returned page.
+
+
+The result is a json with the following structure:
+
+```
+{"modules": [{"module":"github.com/athens-artifacts/no-tags","version":"v1.0.0"}],
+ "next":""}'
+```
+
+An empty nextpagetoken means that no more pages are available.
```

Comment:
an empty continuation token "next" or something like this as `nextpagetoken` is not used
or can we call it continuation token in a response instead of next?

Reply:
I like it as next as it hides implementation details.
Maybe we can omitempty for empty token and not send a token back at all.

Reply:
Agree. I am afraid I left the doc behind with the initial name I used

Reply:
https://github.com/gomods/athens/pull/955/files#diff-becc13561b61d1bfaae2b668bc3aa018R20

This is where we can make the change I believe.